From 874c367720b1875e4025754c44f26201a9cdf28f Mon Sep 17 00:00:00 2001
From: John Youn <johnyoun@synopsys.com>
Date: Wed, 27 Apr 2016 20:20:56 -0700
Subject: [PATCH 194/194] usb: dwc2: Proper cleanup on dr_mode failure
Git-commit: a6ef3e02542a33fb705e6977221deb0292b27398
Patch-mainline: v4.7-rc1
References: bsc#984114

Cleanup in probe if we fail to get dr_mode.

Signed-off-by: John Youn <johnyoun@synopsys.com>
Signed-off-by: Felipe Balbi <felipe.balbi@linux.intel.com>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/usb/dwc2/platform.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc2/platform.c b/drivers/usb/dwc2/platform.c
index 88629be..fc6f525 100644
--- a/drivers/usb/dwc2/platform.c
+++ b/drivers/usb/dwc2/platform.c
@@ -562,7 +562,7 @@ static int dwc2_driver_probe(struct platform_device *dev)
 
 	retval = dwc2_get_dr_mode(hsotg);
 	if (retval)
-		return retval;
+		goto error;
 
 	/*
 	 * Reset before dwc2_get_hwparams() then it could get power-on real
-- 
1.8.5.6

